astrodata/fits.py

```diff
@@ -326,6 +330,95 @@ def normalize_indices(slc, nitems):
 
     return indices, multiple
 
+# Could parametrize some naming conventions in the following two functions if
+# done elsewhere for hard-coded names like 'SCI' in future, but they only have
+# to be self-consistent with one another anyway.
+
+def wcs_to_asdftablehdu(wcs, extver=None):
+    """
+    Serialize a gWCS object as a FITS TableHDU (ASCII) extension.
+
+    The ASCII table is actually a mini ASDF file. The constituent AstroPy
+    models must have associated ASDF "tags" that specify how to serialize them.
+    """
+
+    import jsonschema
+
+    # Create a small ASDF file in memory containing the WCS object
+    # representation because there's no public API for generating only the
+    # relevant YAML subsection and an ASDF file handles the "tags" properly.
+    try:
```

Comment:
Some of this code should probably be spun off into non-FITS-specific routines to support serializing gWCS in other formats, but for now it works.